tests/prometheus_server.go

```diff
@@ -28,25 +32,32 @@ type PrometheusServer struct {
 // NewPrometheusServer creates and starts a Prometheus server with the provided gatherer
 // listening on 127.0.0.1:0 and serving /ext/metrics.
 func NewPrometheusServer(gatherer prometheus.Gatherer) (*PrometheusServer, error) {
+	return NewPrometheusServerWithPort(gatherer, defaultMetricsPort)
+}
+
+// NewPrometheusServerWithPort creates and starts a Prometheus server with the provided gatherer
+// listening on 127.0.0.1:port and serving /ext/metrics.
+func NewPrometheusServerWithPort(gatherer prometheus.Gatherer, port uint64) (*PrometheusServer, error) {
```

Comment:
Consider using functional options pattern here, but think this is fine since it's just adding one extra constructor.